src/main.cpp

```diff
@@ -4210,6 +4217,27 @@ bool ContextualCheckBlock(
         }
     }
 
+    if (!fCheckTransactions)
```

Comment:
This and the remaining block of code could be kept where it originally was but just wrapped in an if (fCheckTransactions) branch.

Reply:
In our pairing you mentioned concern that the subsequent checks might depend upon the correctness of the previous checks. It occurs to me that moving this back to the original location is helpful for the diff, but it doesn't address that problem; I'll try and verify that that's not an issue by inspection.